### PR TITLE
Fix comma-separated medium types parsing in Helm chart

### PR DESCRIPTION
**Pull request by jiacheliu3**
This resolves https://github.com/Alluxio/alluxio/issues/11778

### CODE REVIEW

File: integration/kubernetes/helm-chart/alluxio/templates/_helpers.tpl [hunk 1]
```diff
@@ -166,11 +166,11 @@ resources:
         {{- if contains "," .mediumtype }}
           {{- $type := .type }}
           {{- $path := .path }}
-          {{- $split := split "," .mediumtype }}
-          {{- range $key, $val := $split }}
+          {{- $parts := splitList "," .mediumtype }}
+          {{- range $i, $val := $parts }}
```
Comment (jiacheliu3) [inline integration/kubernetes/helm-chart/alluxio/templates/_helpers.tpl:170]: `split "SSD,HDD"` gives a dict format `{_0: SSD, _1: HDD}`, whereas `splitList "SSD,HDD"` gives `[SSD, HDD]`, which is more intuitive and avoids the underscore replacing in the key.

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Comment by madanadit** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please.

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please.

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please.

**Pull request merged**